### ISSUE TITLE
cmd/cue: def should allow placing exported CUE source into a package

### ISSUE THREAD

**Issue by seh** [cue-lang/cue#2240]
**Is your feature request related to a problem? Please describe.**

When running _cue def_, the generated CUE code is not part of any package. In order to use the code together with other CUE code that does sit within a particular package, I have to edit the generated CUE code to add a package clause.

**Describe the solution you'd like**

Introduce a new command-line flag for the _cue def_ command—perhaps `--within-package`, since the `--package` flag already exists—that induces the command to emit a package clause with the specified package name.

**Describe alternatives you've considered**

Using a shell program, concatenate a package clause (e.g. `package foo\n\n`) and the output of _cue def_ to emulate this capability.

**Additional context**

We discussed this proposal during the Community Call on Tuesday, 31 January 2023 (per #2219). The consensus seemed to be that this omission is unintentional, and that no one had needed the feature badly enough to have implemented it thus far.

Note that the existing `--package` command-line flag specifies the package within which to interpret imported non-CUE sources, and does not influence the package within which the generated and emitted CUE source sits.

I am not sure whether we'd need to ensure that the requested package name via the notional `--within-package` flag does not conflict with any package names imported via import declarations. When used together with the newly arrived `--inline-imports` flag, that collision would likely be immaterial, at least in the emitted CUE code. It might still require some consideration while interpreting the input CUE code before emitting the consolidated form.

**Comment by myitcv** [comment]
Per Slack conversation, this seems totally reasonable. Marking as "Needs planning" for now - we will revisit all feature requests after this big "garden" of open issues.

**Comment by seh** [comment]
Thank you, Paul. I look forward to trying this in a future release.

**Comment by myitcv** [comment]
> Thank you, Paul. I look forward to trying this in a future release.

Great, thank you. It should be part of v0.8.0-alpha.4 today if you fancy trying in a pre-release.

**Comment by myitcv** [comment]
For folks interested in what's possible with this now having landed in [v0.8.0-alpha.4](https://github.com/cue-lang/cue/releases/tag/v0.8.0-alpha.4) here is the [test that landed as part of the change](https://review.gerrithub.io/c/cue-lang/cue/+/1177546/4/cmd/cue/cmd/testdata/script/export_cue.txtar):

```
# Verify that -p works when exporting as CUE.

# Case 1: only non-CUE inputs
exec cue export --out cue -p blah x.json y.json
cmp stdout stdout.golden

# Case 2: CUE input which does not declare a package (with JSON)
exec cue export --out cue -p blah nopkg.cue x.json y.json
cmp stdout stdout.golden

# Case 3: CUE input which does declare same package (with JSON)
exec cue export --out cue -p blah pkg.cue x.json y.json
cmp stdout stdout.golden

# Case 4: CUE input with mis-matched package (with JSON)
#
# TODO: in the future we could allow "repackaging" via --force which would
# cause this export to succeed with the package clause 'blah' in the result.
! exec cue export --out cue -p blah diffpkg.cue x.json y.json
cmp stderr stderr.golden

-- x.json --
{
    "x": 5
}
-- y.json --
{
    "y": 4
}
-- nopkg.cue --
x: 5
y: 4
-- pkg.cue --
package blah

x: 5
y: 4
-- diffpkg.cue --
package other

x: 5
y: 4
-- stdout.golden --
package blah

x: 5
y: 4
-- stderr.golden --
"package" flag clashes with existing package name (blah vs other)
```
